lib/galaxy/datatypes/binary.py

```diff
@@ -4302,3 +4302,28 @@ def display_peek(self, dataset: DatasetProtocol) -> str:
             return dataset.peek
         except Exception:
             return "Binary Numpy npz file (%s)" % (nice_size(dataset.get_size()))
+
+
+@build_sniff_from_prefix
+class FITS(Binary):
+    """
+    FITS (Flexible Image Transport System) file data format, widely used in astronomy
+    Represents sky images (in celestial coordinates) and tables 
+     https://fits.gsfc.nasa.gov/fits_primer.html
+    """
+
+    file_ext = "fits"
+
+    def sniff_prefix(self, file_prefix: FilePrefix):
+        """Determines whether the file is a CASTEP log
+        >>> from galaxy.datatypes.sniff import get_test_fname
+        >>> fname = get_test_fname('legacysurvey.fits')
```

Comment:
Test data needs to be added (in dir `lib/galaxy/datatypes/test/`)

Reply:
added, right @francoismg ?